lib/buildkite.sh

```diff
@@ -383,3 +385,48 @@ EOF
 EOF
   fi
 }
+
+##
+## Generate an annotation for the build based on the plan result
+##
+## This function looks for a plan file at "${build_dir}/terraform.tfplan"
+## If there is no file, we assume the plan failed, and we create an error annotation with a link to the failed job
+## If there is a file, we inspect the .resource_changes[].change.actions fields for each resource.
+## If all of these fields are "no-op", the plan has succeeded with no changes
+## If any of these fields are not "no-op", the plan has succeeded with changes. In this case, we do a terraform show
+## and render these changes in the annotation.
+##
+bk_plan_annotate() {
+  # Ensure that a workspace argument (-w/--workspace) was specified.
+  if [[ -z "${_arg_workspace:-}" ]]; then
+    die "No Terraform workspace specified. This command requires a --workspace argument."
+  fi
+
+  info_msg "Annotating build with plan output"
+
+  _tf_plan_file="${build_dir}/terraform.tfplan"
```

Comment:
I've tried to follow the convention that `_var_name` indicates a file-level variable that is defined outside of function scope (i.e., is probably used by multiple functions within the file). The exception to that is the `_arg_xyz` variables since they get generated by Argbash and have no control over that.

For variables inside functions, it's better to just use `local` so that they don't escape the function scope and skip the leading `_` since it's not necessary.

```suggestion
  local tf_plan_file="${build_dir}/terraform.tfplan"
```

Reply:
Gotcha, makes sense